app/src/main/java/com/firebase/uidemo/RecyclerViewDemoActivity.java

```diff
@@ -71,34 +69,35 @@ public void onComplete(FirebaseError firebaseError, Firebase firebase) {
         mMessages = (RecyclerView) findViewById(R.id.messagesList);
 
         LinearLayoutManager manager = new LinearLayoutManager(this);
-        manager.setStackFromEnd(true);
+        manager.setReverseLayout(false);
 
-        mMessages.setHasFixedSize(true);
+        mMessages.setHasFixedSize(false);
         mMessages.setLayoutManager(manager);
 
-
-        updateChat();
-    }
-
-    protected void updateChat() {
-        FirebaseRecyclerViewAdapter<Chat, ChatHolder> adapter = new FirebaseRecyclerViewAdapter<Chat, ChatHolder>(Chat.class, R.layout.message, ChatHolder.class, mChatRef) {
+        mRecycleViewAdapter = new FirebaseRecyclerViewAdapter<Chat, ChatHolder>(Chat.class, R.layout.message, ChatHolder.class, mChatRef) {
             @Override
             public void populateViewHolder(ChatHolder chatView, Chat chat) {
-
                 chatView.setName(chat.getName());
                 chatView.setText(chat.getText());
 
                 if (mAuthData != null && chat.getUid().equals(mAuthData.getUid())) {
```

Comment:
It's pretty likely that there are chats out there without a `uid`. Will this work with them? If not, it might be worth an extra null check on `chat.getUid()`